ctk/common/src/main/kotlin/org/codice/compliance/utils/TestCommon.kt

```diff
@@ -245,29 +279,137 @@ class TestCommon {
                     .log()
                     .ifValidationFails()
                     .`when`()
-                    .post(Common.getSingleSignOnLocation(SamlProtocol.POST_BINDING))
+                    .post(getSingleSignOnLocation(POST_BINDING))
         }
 
         /**
          * Encodes an AuthnRequest
          * @return A string representation of the encoded input request
          */
-        fun encodeAuthnRequest(authnRequest: AuthnRequest): String {
-            val authnRequestString = authnRequestToString(authnRequest)
+        fun encodeRedirectRequest(request: RequestAbstractType): String {
+            val authnRequestString = samlObjectToString(request)
             authnRequestString.debugPrettyPrintXml(AUTHN_REQUEST)
-            return Encoder.encodeRedirectMessage(authnRequestString)
+            return encodeRedirectMessageWithUrlEncoding(authnRequestString)
         }
 
         /**
          * Signs a given AuthnRequest and converts the object to a String for a POST request.
          * @return The signed AuthnRequest as a String
          */
-        fun signAndEncodeToString(authnRequest: AuthnRequest, relayState: String? = null): String {
-            SimpleSign().signSamlObject(authnRequest)
-            val authnRequestString = authnRequestToString(authnRequest)
+        fun signAndEncodeToString(request: RequestAbstractType,
+            relayState: String? = null): String {
+            SimpleSign().signSamlObject(request)
+            val authnRequestString = samlObjectToString(request)
             authnRequestString.debugPrettyPrintXml(AUTHN_REQUEST)
-            return if (relayState == null) Encoder.encodePostMessage(authnRequestString)
-            else Encoder.encodePostMessage(authnRequestString, relayState)
+            return if (relayState == null) encodePostMessage(authnRequestString)
+            else encodePostMessage(authnRequestString, relayState)
+        }
+
+        /**
+         * Attempts to login to service providers
+         * @param binding - Binding used for login
+         * @param relayState - relay state to send in login
+         * @param singleSP - if true logs in with one sp, else logs in with both
+         * @return cookies from first SP login, to be used in logout request
+         */
+        @Suppress("TooGenericExceptionCaught" /* Catching all Exceptions */)
+        fun loginAndGetCookies(binding: SamlProtocol.Binding, relayState: String? = null,
```

Comment:
I think the only things that need to know there are two SPs are the tests because they start the second SP and login to two SPs. Nothing else should be different.